File: Assignments/10346 Peter's Smokes/10346 PS.py
"""
      Name:   Byron Dowling
      Course: 4883 Programming Techniques
      Date:   11/02/2021

      UVa #10346
      
      - This one was not accepted for time limit, but was good python practice
"""

import sys

line = str(input())

while line != "":
  temp = ""

  try:

    nums = []

    nums = line.split(' ')

    pile = 0
    total = 0
    start_cig_amount = int(nums[0])
    new_cig_amount = int(nums[1])

    while start_cig_amount > 0:
      start_cig_amount -= 1
      pile += 1
      total += 1

    while pile > new_cig_amount:
      total += 1
      pile -= new_cig_amount
      pile += 1

    print(total)

    line = str(input())

  except:
    break

File: Assignments/10346 Peter's Smokes/10346 PS.cpp
/*
    Name:   Byron Dowling
    Course: 4883 Programming Techniques
    Date:   11/02/2021
    
    UVa #10346 Peter's Smokes
*/

#include <iostream>

using namespace std;

typedef long long int LLI;

int main()
{
  long long int x = 0, y = 0;

  while (scanf("%lld %lld", &x, &y) !=EOF) 
  {
    
    LLI total = x;
    LLI remaining = x;

    while (remaining >= y)
    {
      x = remaining / y;                  // Integer division while updating x's value

      remaining = (remaining % y) + x;    // Update remaining

      total += x;                         // Push on total
    }

    printf("%lld\n", total);
  }

  return 0;
}
